addons/mail/static/src/views/activity/activity_model.js

```diff
@@ -0,0 +1,18 @@
+/** @odoo-module */
+
+import { RelationalModel } from "@web/views/relational_model";
+
+export class ActivityModel extends RelationalModel {
+    async load(params = {}) {
+        const activityData = await this.fetchActivityData();
+        await super.load(params);
+        this.activityData = activityData;
+    }
+
+    async fetchActivityData() {
+        return await this.orm.call("mail.activity", "get_activity_data", [], {
+            res_model: this.rootParams.resModel,
+            domain: this.env.searchModel._domain,
```

Comment:
looks fishy to get the domain like that. I think you should take the domain from the load params instead